packages/core/src/tools/codegen/utils.test.ts

```diff
@@ -115,174 +157,239 @@ describe('codegen-utils', () => {
+              './src/tools/codegen/test-data/extract-components/custom-component-map/src/components/TestComponent.tsx'
+            )
           ),
-        ],
-      ]);
+        },
+      ];
+
+      expect(actual).to.deep.equal(expected);
     });
 
-    it('should return imports with absolute paths from component-map.ts with named imports', () => {
+    it('ignores default/named imports and only uses namespace imports (named-imports fixture)', () => {
       const appPath = './src/tools/codegen/test-data/extract-components/named-imports';
-
-      const imports = codegenUtils.resolveComponentImportFiles(appPath);
-      expect(Array.from(imports)).to.deep.equal([
-        [
-          'TestComponent',
-          path.resolve(
-            process.cwd(),
-            './src/tools/codegen/test-data/extract-components/named-imports/src/components/TestComponent.tsx'
+      const res = codegenUtils.resolveComponentImportFiles(appPath);
+
+      expect(res.imports.map((i) => ({ ...i, filePath: norm(i.filePath) }))).to.deep.equal([
+        {
+          componentKey: 'TestComponent',
+          moduleName: 'TestComponent',
+          importPath: '../src/components/TestComponent',
+          filePath: norm(
+            path.resolve(
+              process.cwd(),
+              './src/tools/codegen/test-data/extract-components/named-imports/src/components/TestComponent.tsx'
+            )
           ),
-        ],
-        [
-          'Component1',
-          path.resolve(
-            process.cwd(),
-            './src/tools/codegen/test-data/extract-components/named-imports/src/components/TestComponent2.tsx'
-          ),
-        ],
-        [
-          'Component2',
-          path.resolve(
-            process.cwd(),
-            './src/tools/codegen/test-data/extract-components/named-imports/src/components/TestComponent2.tsx'
-          ),
-        ],
+        },
       ]);
     });
 
-    it('should return imports with absolute paths from component-map.ts when paths aliases are used', () => {
+    it('resolves with path aliases defined in tsconfig (with-path-aliases fixture)', () => {
       const appPath = './src/tools/codegen/test-data/extract-components/with-path-aliases';
-      const imports = codegenUtils.resolveComponentImportFiles(appPath);
+      const res = codegenUtils.resolveComponentImportFiles(appPath);
 
-      expect(Array.from(imports)).to.deep.equal([
-        [
-          'TestComponent',
+      expect(res.uniqueFiles.map(norm)).to.deep.equal([
+        norm(
           path.resolve(
             process.cwd(),
             './src/tools/codegen/test-data/extract-components/with-path-aliases/src/components/TestComponent.tsx'
-          ),
-        ],
+          )
+        ),
       ]);
     });
 
-    it('should ignore imports starting with "node:", ending with ".d.ts" and containing "node_modules"', () => {
+    it('ignores non-relevant imports (node:, *.d.ts, node_modules) in the map entries', () => {
```

Comment:
This test passes because node_modules import is not being used in test component map.
If you add
```
  ['fs', fs],
  ['index', index],
```
in component map init under node-modules-imports, it will fail